### PR TITLE
Better plotly

### LINKED ISSUE #310: Unify image and slice viewer
Currently viewing a 2D image and a slice through 3D data use two different functions:
`plot_image` and `plot_sliceviewer`
One should unify the code so that the 2D image viewer is just a special case of the `sliceviewer` with no sliders.

### PR DESCRIPTION
**Pull request by nvaytet**
Update to the `plotly` plotting to more general methods.

- The 2D image is now a special case of the N-D viewer that happens to not have any sliders.
- There are now buttons to switch axes that are plotted
- There is a 3D slicer which can be enabled via the `projection='3d'` keyword argument.
- Some changes to the `travis` config were made so we could install `plotly>=4.0.0`. We switched from using `pip` to using `conda` for installing Python requirements.

Demo:
```Python
import numpy as np
import scipp as sc
from scipp import Dim
N = 50
M = 40
L = 30
K = 20
xx = np.arange(N, dtype=np.float64) + 2.0
yy = np.arange(M, dtype=np.float64)
zz = np.arange(L, dtype=np.float64)
kk = np.arange(K, dtype=np.float64)
k, z, y, x = np.meshgrid(kk, zz, yy, xx, indexing='ij')
# print(np.shape(x))
b = N/20.0
c = M/2.0
d = L/2.0
r = np.sqrt(((x-c)/b)**2 + ((y-c)/b)**2 + ((z-d)/b)**2 + k)
a = np.sin(r)
d3 = sc.Dataset()
d3.coords[Dim.X] = sc.Variable([Dim.X], values=xx)
d3.coords[Dim.Y] = sc.Variable([Dim.Y], values=yy)
d3.coords[Dim.Z] = sc.Variable([Dim.Z], values=zz)
d3.coords[Dim.Qx] = sc.Variable([Dim.Qx], values=kk)
d3['Some3Ddata'] = sc.Variable([Dim.Qx, Dim.Z, Dim.Y, Dim.X], values=a, variances=0.1*a)
sc.plot(d3, backend="plotly")
```
Screenshot at 2019-09-24 23-06-10
```Python
import numpy as np
import scipp as sc
from scipp import Dim
N = 50
M = 40
L = 30
K = 20
xx = np.arange(N, dtype=np.float64) + 2.0
yy = np.arange(M, dtype=np.float64)
zz = np.arange(L, dtype=np.float64)
kk = np.arange(K, dtype=np.float64)
k, z, y, x = np.meshgrid(kk, zz, yy, xx, indexing='ij')
# print(np.shape(x))
b = N/20.0
c = M/2.0
d = L/2.0
r = np.sqrt(((x-c)/b)**2 + ((y-c)/b)**2 + ((z-d)/b)**2 + k)
a = np.sin(r)
d3 = sc.Dataset()
d3.coords[Dim.X] = sc.Variable([Dim.X], values=xx)
d3.coords[Dim.Y] = sc.Variable([Dim.Y], values=yy)
d3.coords[Dim.Z] = sc.Variable([Dim.Z], values=zz)
d3.coords[Dim.Qx] = sc.Variable([Dim.Qx], values=kk)
d3['Some3Ddata'] = sc.Variable([Dim.Qx, Dim.Z, Dim.Y, Dim.X], values=a, variances=0.1*a)
sc.plot(d3, backend="plotly", projection='3d')
```
Screenshot at 2019-09-24 23-05-23

Note that the fancy zooming and sub-sampling of large images has been removed for now. It added much complicated code, and the performance was not ideal. We will look at other options, including the Datashader library, to fix this.

Fixes #310 #302

### CODE REVIEW

**Comment by nvaytet** [comment]
@DanNixon I made some changes to the `travis` config, including moving around some `conda` stuff. Can you check that I did not do something horribly wrong? I'm basically using `conda` to install the python dependencies, because I need `plotly >=4` and I needed to upgrade `pip` with that, and upgrading `pip` on travis seems almost impossible.

File: .travis.yml [hunk 1]
```diff
@@ -28,22 +28,32 @@ jobs:
       compiler: gcc
       name: "gcc"
       if: NOT type = cron
-      before_install: .travis/before_install.sh
+      env:
+        - PATH="$HOME/miniconda/bin:$PATH"
```
Comment (DanNixon) [inline .travis.yml]: If this is now needed everywhere could you move it to the `global` section above (same for all other stages).

**Comment by nvaytet** [comment]
@SimonHeybrock @DanNixon note that I changed the distribution to be the Ubuntu `bionic` (18.04) since we all seem to be developing on this version anyway.

**Comment by SimonHeybrock** [comment]
I am a bit worried about installing miniconda for all builds? Doesn't this significantly increase the time for each and every build we run?

**Comment by SimonHeybrock** [comment]
Does the newer Ubutnu version remove the need to use conda for plotly?

**Comment by DanNixon** [comment]
> I am a bit worried about installing miniconda for all builds? Doesn't this significantly increase the time for each and every build we run?

Probably.

If we wanted to do this we'd be better off using a containerised build environment.
I did a similar thing for https://github.com/DMSC-Instrument-Data/MDSpaceFillingPrototype since getting the correct compilers on Travis was a pain.

This would save us a fair bit of time over even what we have now.

**Comment by nvaytet** [comment]
> @SimonHeybrock @DanNixon note that I changed the Travis distribution to be the Ubuntu `bionic` (18.04) since we all seem to be developing on this version anyway.
> 
> Awesome that this is now available (apparently since June?)! Does it make the setup simpler (fewer hoops to get correct compilers and libs?)?

Yes fewer loops. Maybe as you suggested it also helps with a newer version of `pip`. I will try and put everything back as it was, but just use the `bionic` distrib

**Comment by nvaytet** [comment]
@SimonHeybrock @DanNixon I removed the extrac `conda` stuff and am now just using `pip` from the `bionic` distribution, which seems to work. Could you have a final look at this? Cheers

**Pull request merged**